# AFRICAN STARTUPS' FUNDING

We are going to use an African startup dataset in an attempt to answer question(s) about African Startups.

**Importing important libraries for numeric manipulation and data viz**

In [56]:
#import all the important libraries
import pandas as pd
import numpy as  np
import matplotlib.pyplot as plt

**Loading the dataset, it is a .xlsx file**

In [57]:
#file location 
african_startups = r"C:\Users\user\Downloads\AfricanStartups\2021\Funded African tech startups 2021.xlsx"
#read and assign the xlsx to a df
df = pd.read_excel(african_startups)

In [58]:
#view df
df.head()

Startup  Unnamed: 1 Location         Sector                Date  \
0  Daystar Power         NaN  Nigeria         energy      January + July   
1        uLesson         NaN  Nigeria        ed-tech  January + December   
2      Cowrywise         NaN  Nigeria        fintech             January   
3         Ogelle         NaN  Nigeria  entertainment             January   
4         Contro         NaN  Nigeria       e-health             January   

   Amount (US$)                                          Investors  \
0    62000000.0  Investment Fund for Developing Countries (IFU)...   
1    22500000.0  Owl Ventures, TLcom Capital, Founder Collectiv...   
2     3000000.0  Quona Capital, Sahil Lavingia, Tsadik Foundati...   
3           NaN                            African Discovery Group   
4       42000.0                                                NaN   

                 Stage  Debt/Equity Female co-founder/CEO? Local founder?  \
0             Series B  Equity/debt                     No            Yes   
1  Series A / Series B       Equity                     No            Yes   
2         Pre-Series A       Equity                     No            Yes   
3          Undisclosed       Equity                     No            Yes   
4             Pre-seed       Equity                     No            Yes   

  Intnl founder?  Launched? Accelerator?  Employees  
0            Yes     2017.0           No       71.0  
1             No     2019.0           No       83.0  
2             No     2017.0          Yes       96.0  
3             No     2017.0           No        7.0  
4             No     2020.0           No        9.0

Based on the initial inspection of the dataset, we need to drop the  **'Unnamed: 1'** column.

In [59]:
df = df.drop(['Unnamed: 1'], axis=1) #removing the unnamed column from above, it has empty values
df.head(5)

Startup Location         Sector                Date  Amount (US$)  \
0  Daystar Power  Nigeria         energy      January + July    62000000.0   
1        uLesson  Nigeria        ed-tech  January + December    22500000.0   
2      Cowrywise  Nigeria        fintech             January     3000000.0   
3         Ogelle  Nigeria  entertainment             January           NaN   
4         Contro  Nigeria       e-health             January       42000.0   

                                           Investors                Stage  \
0  Investment Fund for Developing Countries (IFU)...             Series B   
1  Owl Ventures, TLcom Capital, Founder Collectiv...  Series A / Series B   
2  Quona Capital, Sahil Lavingia, Tsadik Foundati...         Pre-Series A   
3                            African Discovery Group          Undisclosed   
4                                                NaN             Pre-seed   

   Debt/Equity Female co-founder/CEO? Local founder? Intnl founder?  \
0  Equity/debt                     No            Yes            Yes   
1       Equity                     No            Yes             No   
2       Equity                     No            Yes             No   
3       Equity                     No            Yes             No   
4       Equity                     No            Yes             No   

   Launched? Accelerator?  Employees  
0     2017.0           No       71.0  
1     2019.0           No       83.0  
2     2017.0          Yes       96.0  
3     2017.0           No        7.0  
4     2020.0           No        9.0

# EDA

In [60]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1173 entries, 0 to 1172
Data columns (total 14 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Startup                 564 non-null    object 
 1   Location                564 non-null    object 
 2   Sector                  564 non-null    object 
 3   Date                    457 non-null    object 
 4   Amount (US$)            310 non-null    float64
 5   Investors               545 non-null    object 
 6   Stage                   564 non-null    object 
 7   Debt/Equity             564 non-null    object 
 8   Female co-founder/CEO?  564 non-null    object 
 9   Local founder?          564 non-null    object 
 10  Intnl founder?          564 non-null    object 
 11  Launched?               564 non-null    float64
 12  Accelerator?            564 non-null    object 
 13  Employees               565 non-null    float64
dtypes: float64(3), object(11)
memory usage: 

Lets check the tail of this dataset

In [61]:
df.tail()

Startup Location Sector Date  Amount (US$) Investors Stage Debt/Equity  \
1168     NaN      NaN    NaN  NaN           NaN       NaN   NaN         NaN   
1169     NaN      NaN    NaN  NaN           NaN       NaN   NaN         NaN   
1170     NaN      NaN    NaN  NaN           NaN       NaN   NaN         NaN   
1171     NaN      NaN    NaN  NaN           NaN       NaN   NaN         NaN   
1172     NaN      NaN    NaN  NaN           NaN       NaN   NaN         NaN   

     Female co-founder/CEO? Local founder? Intnl founder?  Launched?  \
1168                    NaN            NaN            NaN        NaN   
1169                    NaN            NaN            NaN        NaN   
1170                    NaN            NaN            NaN        NaN   
1171                    NaN            NaN            NaN        NaN   
1172                    NaN            NaN            NaN        NaN   

     Accelerator?  Employees  
1168          NaN        NaN  
1169          NaN        NaN  
1170          NaN        NaN  
1171          NaN        NaN  
1172          NaN        0.0

We have around 564 Non-null values for 1173 rows. We can assume that after row 564, we have NULL value. We'll have to investigate this. 

In [62]:
df.iloc[[560,561,562,563,564,565,566,567]]

Startup  Location       Sector     Date  Amount (US$)  \
560  OKO Finance      Mali      fintech    April     1200000.0   
561       alsoug     Sudan   e-commerce  October     5000000.0   
562       Ramani  Tanzania  retail-tech      NaN           NaN   
563        Gozem      Togo    transport  October     5000000.0   
564          NaN       NaN          NaN      NaN           NaN   
565          NaN       NaN          NaN      NaN           NaN   
566          NaN       NaN          NaN      NaN           NaN   
567          NaN       NaN          NaN      NaN           NaN   

                                             Investors        Stage  \
560  Newfund, ResiliAnce, Mercy Corps Ventures, Tec...         Seed   
561  Fawry, Rally Cap Ventures, Musha Ventures, others     Series A   
562             Launch Africa Ventures, Musha Ventures  Undisclosed   
563  AAIC, Thunes (TransferTo), Momentum Ventures (...     Series A   
564                                                NaN          NaN   
565                                                NaN          NaN   
566                                                NaN          NaN   
567                                                NaN          NaN   

    Debt/Equity Female co-founder/CEO? Local founder? Intnl founder?  \
560      Equity                     No             No            Yes   
561      Equity                    Yes            Yes             No   
562      Equity                     No            Yes             No   
563      Equity                     No            Yes             No   
564         NaN                    NaN            NaN            NaN   
565         NaN                    NaN            NaN            NaN   
566         NaN                    NaN            NaN            NaN   
567         NaN                    NaN            NaN            NaN   

     Launched? Accelerator?  Employees  
560     2017.0          Yes       12.0  
561     2016.0           No       11.0  
562     2019.0          Yes       55.0  
563     2018.0           No      114.0  
564        NaN          NaN        NaN  
565        NaN          NaN        NaN  
566        NaN          NaN        NaN  
567        NaN          NaN        NaN

The above dataset proves our earlier suspicion that our data has **564** rows instead of the **1173** rows.

**We need to select a subset of ROWS from 0-563** before we continue..

### subset with 564 rows

In [63]:
subset_df = df.iloc[0:564] #the range is not inclusive of the 564th row.
subset_df.tail() #checking if the last row has Nulls or not

Startup  Location       Sector     Date  Amount (US$)  \
559  AlphaDirect  Botswana      fintech    April      600000.0   
560  OKO Finance      Mali      fintech    April     1200000.0   
561       alsoug     Sudan   e-commerce  October     5000000.0   
562       Ramani  Tanzania  retail-tech      NaN           NaN   
563        Gozem      Togo    transport  October     5000000.0   

                                             Investors         Stage  \
559  Launch Africa Ventures, Century Oak Capital, a...  Pre-Series A   
560  Newfund, ResiliAnce, Mercy Corps Ventures, Tec...          Seed   
561  Fawry, Rally Cap Ventures, Musha Ventures, others      Series A   
562             Launch Africa Ventures, Musha Ventures   Undisclosed   
563  AAIC, Thunes (TransferTo), Momentum Ventures (...      Series A   

    Debt/Equity Female co-founder/CEO? Local founder? Intnl founder?  \
559      Equity                     No             No            Yes   
560      Equity                     No             No            Yes   
561      Equity                    Yes            Yes             No   
562      Equity                     No            Yes             No   
563      Equity                     No            Yes             No   

     Launched? Accelerator?  Employees  
559     2015.0           No        4.0  
560     2017.0          Yes       12.0  
561     2016.0           No       11.0  
562     2019.0          Yes       55.0  
563     2018.0           No      114.0

#### Checking for nulls again

In [64]:
subset_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 564 entries, 0 to 563
Data columns (total 14 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Startup                 564 non-null    object 
 1   Location                564 non-null    object 
 2   Sector                  564 non-null    object 
 3   Date                    457 non-null    object 
 4   Amount (US$)            310 non-null    float64
 5   Investors               545 non-null    object 
 6   Stage                   564 non-null    object 
 7   Debt/Equity             564 non-null    object 
 8   Female co-founder/CEO?  564 non-null    object 
 9   Local founder?          564 non-null    object 
 10  Intnl founder?          564 non-null    object 
 11  Launched?               564 non-null    float64
 12  Accelerator?            564 non-null    object 
 13  Employees               564 non-null    float64
dtypes: float64(3), object(11)
memory usage: 61

columns with missing valies are date, amount, inventors

**Lets investigate**

In [65]:
subset_df.head(10)

Startup Location         Sector                Date  Amount (US$)  \
0    Daystar Power  Nigeria         energy      January + July    62000000.0   
1          uLesson  Nigeria        ed-tech  January + December    22500000.0   
2        Cowrywise  Nigeria        fintech             January     3000000.0   
3           Ogelle  Nigeria  entertainment             January           NaN   
4           Contro  Nigeria       e-health             January       42000.0   
5             Xend  Nigeria        fintech            February      500000.0   
6      Oneport 365  Nigeria      logistics            February           NaN   
7           Tǽillὸ  Nigeria     e-commerce            February           NaN   
8         ImaliPay  Nigeria        fintech            February           NaN   
9  Get It Done Now  Nigeria     e-commerce            February           NaN   

                                           Investors                    Stage  \
0  Investment Fund for Developing Countries (IFU)...                 Series B   
1  Owl Ventures, TLcom Capital, Founder Collectiv...      Series A / Series B   
2  Quona Capital, Sahil Lavingia, Tsadik Foundati...             Pre-Series A   
3                            African Discovery Group              Undisclosed   
4                                                NaN                 Pre-seed   
5                              NGC Ventures, Hashkey  Undisclosed (strategic)   
6                                    CcHub Syndicate              Undisclosed   
7                                    CcHub Syndicate              Undisclosed   
8  TEN13, FINCA Ventures, Optimiser Foundation, M...                 Pre-seed   
9                                                NaN              Undisclosed   

   Debt/Equity Female co-founder/CEO? Local founder? Intnl founder?  \
0  Equity/debt                     No            Yes            Yes   
1       Equity                     No            Yes             No   
2       Equity                     No            Yes             No   
3       Equity                     No            Yes             No   
4       Equity                     No            Yes             No   
5       Equity                     No            Yes             No   
6       Equity                     No            Yes             No   
7       Equity                    Yes            Yes             No   
8       Equity                     No            Yes             No   
9       Equity                     No            Yes            Yes   

   Launched? Accelerator?  Employees  
0     2017.0           No       71.0  
1     2019.0           No       83.0  
2     2017.0          Yes       96.0  
3     2017.0           No        7.0  
4     2020.0           No        9.0  
5     2019.0          Yes       68.0  
6     2019.0           No       17.0  
7     2018.0          Yes       12.0  
8     2020.0           No        7.0  
9     2018.0           No        4.0

Most of the null values in the amount column, they're undisclosed amount. Should I just replace them with **ZEROs**

In [66]:
subset_df['Amount (US$)'].isnull().sum() # we should fill this with Zeros and how many of these are undisclosed

254

In [67]:
subset_df['Stage'].value_counts()

Undisclosed                               236
Seed                                      123
Pre-seed                                   96
Series A                                   41
Pre-Series A                               21
Undisclosed                                16
Series B                                   11
Series C                                    3
Pre-Series B                                2
Bridge                                      2
Seed + Pre-Series A                         1
Pe-seed                                     1
Strategic                                   1
Seed / Pre-Series A                         1
Series A                                    1
Pre-seed / Seed                             1
Pre-Series C                                1
Seed/bridge                                 1
Series A / Series B                         1
Seed / Series A                             1
Series A / B                                1
Undisclosed (strategic)           

I somewhat went of on a tangent investigate how many startup disclose their stage of investment and is it equal to the number of Nulls value. Found that it is not the same but rather close.

## AMOUNT (US) COLUMN

#### fill the NaN with zeros.

In [68]:
subset_df['Amount (US$)'].fillna(0, inplace=True) #replacing the NaN with Zeros, had to use the INPLACE = TRUE didnt work without it

C:\Users\user\AppData\Local\Temp\ipykernel_12260\854645118.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  subset_df['Amount (US$)'].fillna(0, inplace=True) #replacing the NaN with Zeros, had to use the INPLACE = TRUE didnt work without it


In [69]:
subset_df.head()

Startup Location         Sector                Date  Amount (US$)  \
0  Daystar Power  Nigeria         energy      January + July    62000000.0   
1        uLesson  Nigeria        ed-tech  January + December    22500000.0   
2      Cowrywise  Nigeria        fintech             January     3000000.0   
3         Ogelle  Nigeria  entertainment             January           0.0   
4         Contro  Nigeria       e-health             January       42000.0   

                                           Investors                Stage  \
0  Investment Fund for Developing Countries (IFU)...             Series B   
1  Owl Ventures, TLcom Capital, Founder Collectiv...  Series A / Series B   
2  Quona Capital, Sahil Lavingia, Tsadik Foundati...         Pre-Series A   
3                            African Discovery Group          Undisclosed   
4                                                NaN             Pre-seed   

   Debt/Equity Female co-founder/CEO? Local founder? Intnl founder?  \
0  Equity/debt                     No            Yes            Yes   
1       Equity                     No            Yes             No   
2       Equity                     No            Yes             No   
3       Equity                     No            Yes             No   
4       Equity                     No            Yes             No   

   Launched? Accelerator?  Employees  
0     2017.0           No       71.0  
1     2019.0           No       83.0  
2     2017.0          Yes       96.0  
3     2017.0           No        7.0  
4     2020.0           No        9.0

In [70]:
#investigate NaN values in the column
subset_df['Amount (US$)'].isnull().sum() # we have removed all nulls

0

### THE investor column

In [71]:
subset_df['Investors'].head()

0    Investment Fund for Developing Countries (IFU)...
1    Owl Ventures, TLcom Capital, Founder Collectiv...
2    Quona Capital, Sahil Lavingia, Tsadik Foundati...
3                              African Discovery Group
4                                                  NaN
Name: Investors, dtype: object

I want to split the investor column into multiple columns.

In [72]:
investor_df = subset_df['Investors'].str.split(',', expand=True)

In [74]:
investor_df.head()

0                1   \
0  Investment Fund for Developing Countries (IFU)             STOA   
1                                    Owl Ventures    TLcom Capital   
2                                   Quona Capital   Sahil Lavingia   
3                         African Discovery Group             None   
4                                             NaN              NaN   

                    2                                      3               4   \
0             Proparco   Morgan Stanley Investment Management   DEG (January)   
1   Founder Collective                             LocalGlobe            None   
2    Tsadik Foundation                                 angels            None   
3                 None                                   None            None   
4                  NaN                                    NaN             NaN   

            5       6     7     8     9   ...    14    15    16    17    18  \
0   IFC (July)   Trine  None  None  None  ...  None  None  None  None  None   
1         None    None  None  None  None  ...  None  None  None  None  None   
2         None    None  None  None  None  ...  None  None  None  None  None   
3         None    None  None  None  None  ...  None  None  None  None  None   
4          NaN     NaN   NaN   NaN   NaN  ...   NaN   NaN   NaN   NaN   NaN   

     19    20    21    22    23  
0  None  None  None  None  None  
1  None  None  None  None  None  
2  None  None  None  None  None  
3  None  None  None  None  None  
4   NaN   NaN   NaN   NaN   NaN  

[5 rows x 24 columns]

In [75]:
investor_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 564 entries, 0 to 563
Data columns (total 24 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   0       545 non-null    object
 1   1       272 non-null    object
 2   2       190 non-null    object
 3   3       140 non-null    object
 4   4       105 non-null    object
 5   5       78 non-null     object
 6   6       58 non-null     object
 7   7       41 non-null     object
 8   8       34 non-null     object
 9   9       27 non-null     object
 10  10      21 non-null     object
 11  11      16 non-null     object
 12  12      13 non-null     object
 13  13      10 non-null     object
 14  14      8 non-null      object
 15  15      7 non-null      object
 16  16      5 non-null      object
 17  17      4 non-null      object
 18  18      3 non-null      object
 19  19      2 non-null      object
 20  20      2 non-null      object
 21  21      2 non-null      object
 22  22      1 non-null      ob

Splitting the investor column will complicate things. An alternative is to count the number of investors.

#### OPTION 2: 

In [76]:
subset_df['Investors_no'] = subset_df['Investors'].str.count(',') + 1 #the +1 is because py indexes from 0

C:\Users\user\AppData\Local\Temp\ipykernel_12260\3275451116.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  subset_df['Investors_no'] = subset_df['Investors'].str.count(',') + 1


In [77]:
subset_df.head()

Startup Location         Sector                Date  Amount (US$)  \
0  Daystar Power  Nigeria         energy      January + July    62000000.0   
1        uLesson  Nigeria        ed-tech  January + December    22500000.0   
2      Cowrywise  Nigeria        fintech             January     3000000.0   
3         Ogelle  Nigeria  entertainment             January           0.0   
4         Contro  Nigeria       e-health             January       42000.0   

                                           Investors                Stage  \
0  Investment Fund for Developing Countries (IFU)...             Series B   
1  Owl Ventures, TLcom Capital, Founder Collectiv...  Series A / Series B   
2  Quona Capital, Sahil Lavingia, Tsadik Foundati...         Pre-Series A   
3                            African Discovery Group          Undisclosed   
4                                                NaN             Pre-seed   

   Debt/Equity Female co-founder/CEO? Local founder? Intnl founder?  \
0  Equity/debt                     No            Yes            Yes   
1       Equity                     No            Yes             No   
2       Equity                     No            Yes             No   
3       Equity                     No            Yes             No   
4       Equity                     No            Yes             No   

   Launched? Accelerator?  Employees  Investors_no  
0     2017.0           No       71.0           7.0  
1     2019.0           No       83.0           4.0  
2     2017.0          Yes       96.0           4.0  
3     2017.0           No        7.0           1.0  
4     2020.0           No        9.0           NaN

### changing the datatype to int
#### launched? column

In [87]:
subset_df['Launched?'] = subset_df['Launched?'].astype(int) #converting the Launched column to an int
subset_df['Employees'] = subset_df['Employees'].astype(int)#employees column
subset_df['Investors_no'].fillna(0, inplace=True)
subset_df['Investors_no'] = subset_df['Investors_no'].astype(int) #number of investors

C:\Users\user\AppData\Local\Temp\ipykernel_12260\2191712135.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  subset_df['Launched?'] = subset_df['Launched?'].astype(int) #converting the Launched column to an int
C:\Users\user\AppData\Local\Temp\ipykernel_12260\2191712135.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  subset_df['Employees'] = subset_df['Employees'].astype(int)#employees column
C:\Users\user\AppData\Local\Temp\ipykernel_12260\2191712135.py:3: SettingWithCopyWarning: 
A value is trying 

In [88]:
subset_df.head()

Startup Location         Sector                Date  Amount (US$)  \
0  Daystar Power  Nigeria         energy      January + July    62000000.0   
1        uLesson  Nigeria        ed-tech  January + December    22500000.0   
2      Cowrywise  Nigeria        fintech             January     3000000.0   
3         Ogelle  Nigeria  entertainment             January           0.0   
4         Contro  Nigeria       e-health             January       42000.0   

                                           Investors                Stage  \
0  Investment Fund for Developing Countries (IFU)...             Series B   
1  Owl Ventures, TLcom Capital, Founder Collectiv...  Series A / Series B   
2  Quona Capital, Sahil Lavingia, Tsadik Foundati...         Pre-Series A   
3                            African Discovery Group          Undisclosed   
4                                                NaN             Pre-seed   

   Debt/Equity Female co-founder/CEO? Local founder? Intnl founder?  \
0  Equity/debt                     No            Yes            Yes   
1       Equity                     No            Yes             No   
2       Equity                     No            Yes             No   
3       Equity                     No            Yes             No   
4       Equity                     No            Yes             No   

   Launched? Accelerator?  Employees  Investors_no  
0       2017           No         71             7  
1       2019           No         83             4  
2       2017          Yes         96             4  
3       2017           No          7             1  
4       2020           No          9             0

## TASK 1:What is the right time to seek funding?


## TASK 2: Do women-led startups get much less funding than men?

## TASK 3:Which sector is funded the most?

## TASK 4:What is the average age of these startups?

## TASK 5:Do accelerators increase the chances of funding and the amount?

## TASK 6:Which investors spends the most amount of money in Africa?
#### this can be broken down into country and  regions